web-src/src/components/Editor.js

```diff
@@ -123,11 +127,18 @@ function Editor({ setResults }) {
         <ParametersView expressions={expressions} state={parameters} setState={setParameters} />
       </Flex>
       <Flex direction="row" gap="size-400" gridColumn='span 2' alignItems="center">
-        <Button variant="primary" isPending={pending} onPress={completionHandler} isDisabled={pending}>
-          <GenerateIcon/>
-          <Text>Generate</Text>
+        <Button
+          UNSAFE_className="hover-cursor-pointer"
+          width="size-1700"
+          variant="primary"
+          style="fill"
+          onPress={completionHandler}
+          isDisabled={pending}>
+            {pending ? <ProgressCircle size="S" aria-label="Generate" isIndeterminate right="10px"/> : <SenseiGenAIIcon />}
```

Comment:
The button from the React Spectrum library has a built-in pending state indicator. Should we use it instead?